src/plugins/crypto.c

```diff
@@ -462,6 +506,68 @@ gchar* bd_crypto_luks_status (const gchar *luks_device, GError **error) {
     return ret;
 }
 
+#ifdef LIBCRYPTSETUP_2
+static struct crypt_pbkdf_type *get_pbkdf_params (BDCryptoLUKSPBKDF *user_pbkdf, GError **error) {
+    const struct crypt_pbkdf_type *default_pbkdf = NULL;
+    struct crypt_pbkdf_type *new_pbkdf = NULL;
+
+    if (user_pbkdf == NULL || user_pbkdf->type == NULL)
+        return NULL;
+
+    /* crypt_get_pbkdf_default returns default pbkdf parameters only based
+       on the luks version -- so for LUKS2 it returns default values for
+       argon2 but we also need to be able to provide default values if user
+       wants pbkdf2 and only specifies type -- we will use the defaults for
+       argon2 and ignore parmaters specific for it
+       better API for this should be part of cryptsetup 2.0.4
+     */
+    default_pbkdf = crypt_get_pbkdf_default (CRYPT_LUKS2);
+    if (!default_pbkdf) {
+        g_set_error (error, BD_CRYPTO_ERROR, BD_CRYPTO_ERROR_FORMAT_FAILED,
+                     "Failed to get default values for pbkdf.");
+        return NULL;
+    }
+
+    new_pbkdf = g_new0 (struct crypt_pbkdf_type, 1);
+    new_pbkdf->type = user_pbkdf->type;
+
+    if (user_pbkdf->hash)
+        new_pbkdf->hash = user_pbkdf->hash;
+    else
+        new_pbkdf->hash = default_pbkdf->hash;
+
+    if (user_pbkdf->time_ms)
+        new_pbkdf->time_ms = user_pbkdf->time_ms;
+    else
+        new_pbkdf->time_ms = default_pbkdf->time_ms;
+
+    if (user_pbkdf->iterations)
+        new_pbkdf->iterations = user_pbkdf->iterations;
+    else
+        new_pbkdf->iterations = default_pbkdf->iterations;
+
+    /* 'max_memory_kb' and 'parallel_threads' are not used in pbkdf2 */
+    if (g_strcmp0 (user_pbkdf->type, CRYPT_KDF_PBKDF2) == 0) {
+        if (user_pbkdf->max_memory_kb)
+            bd_utils_log (LOG_WARNING, "'max_memory_kb' is not valid option for 'pbkdf2', ignoring.");
+
+        new_pbkdf->max_memory_kb = 0;
+        new_pbkdf->parallel_threads = 0;
```

Comment:
Why is ``parallel_threads`` not supported by ``BDCryptoLUKSPBKDF``? Is it just a matter of time somebody asks for this?